### ISSUE TITLE
Unexpected error en registration process

### ISSUE THREAD

**Issue by sergiokessler** [mimecorg/webissues#44]
Hi there,

how I can debug this ?

the message says me absolutely nothing at all, 
also, I see nothing in the server log files...

![image](https://user-images.githubusercontent.com/14111329/126691583-91cf9f2d-49b4-411c-9ae5-b4f7d9d17f5d.png)

**Comment by mimecorg** [comment]
When an error like this occurs, you should be able to see the detailed information in the event log (Tools > Event Log).

**Comment by sergiokessler** [comment]
Ok, I found it, thanks !

`Unhandled PHPMailer\PHPMailer\Exception: Error SMTP ...`

Maybe I misconfigured the mail settings, but the problem is that accounts created remains in an inconsistent state...
So, I think this is a bug...

(you don't see any in User Accounts but trying to re-register fail with 'username already exist')

**Comment by sergiokessler** [comment]
ok, `delete * from register_requests` cleaned up things...

**Comment by sergiokessler** [comment]
Now, there is 2 additional problems:

Mail setings:
![image](https://user-images.githubusercontent.com/14111329/126817027-846b09d1-c45a-400e-ac77-ea3a12e711b7.png)

1) this doesn't produce any log at all

2) the PHPMailer included is old, very old...

**Comment by mimecorg** [comment]
Yes, you're right, the error should be logged when email settings are tested, I will fix that. PHPMailer is from December 2019, so it's not super old, but I will update it in the next release of WebIssues.

Requests that are not activated are automatically removed from the register_requests table by the cron job. It's not perfect, but even if the email is sent, there is no guarantee that it reaches the recipient.

**Comment by sergiokessler** [comment]
ok, thanks!

IMO the `register_requests` records should not be stored at all if there is an error in the process...
(like that "unexpected error")

do you have an estimate when the next release is coming ?

**Comment by gh658804** [comment]
In a somewhat related issue, i had the webissues database taken offline and i received the same error. This error is too general. It would be nice if it could be analyzed a bit. Perhaps something like "Unexpected error in * subsystem", where * could be Mail, Database or whatever else could exist. At least then you could narrow it down and immediately know where to look without checking any logs.

BTW, "Tools > Event Log" as said in your first post was not helpful for me. Perhaps the message could also have a more specific directions on where to go to find a more detailed description of the error.

**Comment by hellmediadev** [comment]
In my case the view "effective_rights" was missing after a migration to another server. The error handling / logging is not useful at all in such cases.

**Comment by geri777** [comment]
In my case the view "effective_rights" was missing after a migration to another server. The error handling / logging is not useful at all in such cases.
